### PR TITLE
Ajouter le swagger à l'API

### LINKED ISSUE #23: Swagger for documentation

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #23

### CODE REVIEW

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 1]
```diff
@@ -0,0 +1,381 @@
+info:
+  title: API Recherche d’entreprises
+  version: 1.0.0
+  contact:
+    email: recherche-entreprises@api.gouv.fr
+    name: Contact API Recherche d’entreprises
+    url: https://recherche-entreprises.api.gouv.fr/
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:7]: We can use https://stats.uptimerobot.com/kXzwzfk7BE

Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:7]: We need to add a redirect https://recherche-entreprises.api.gouv.fr/ -> https://api.gouv.fr/api-recherche-entreprises

**Review by XavierJp — APPROVED** [review]

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  license:
+    name: GNU Affero General Public License v3.0
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:9]: Why not MIT ? @XavierJp

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+              schema:
+                type: object
+                properties:
+                  results:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        siren:
+                          type: integer
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:132]: Siren should be string

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                          example: 111111111
+                          description: le numéro unique de l'entreprise
+                        siret_siege:
+                          type: integer
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:136]: siret should be string

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+                        date_creation_siege:
+                          type: string
+                          format: date-time
+                          description: Date de création du siège
+                        tranche_effectif_salarie_siege:
+                          type: string
+                        activite_principale_registre_metier:
+                          type: string
+                        numero_voie:
+                          type: number
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:148]: Are you sure ? What about "23 bis rue de ségur" ? `numero_voie` is `23` or `23 bis`. I think there a specific column on sirene database for this info that I do not find in this swagger, it is something like : `indicederepetition`

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+                type: object
+                properties:
+                  results:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        siren:
+                          type: integer
+                          example: 111111111
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:293]: same here than above. Same for siret and type_voie fields

**Review by geoffreyaldebert — CHANGES_REQUESTED** [review]

File: aio/aio-proxy/aio_proxy/doc/open-api.yml (hunk 3 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/doc/open-api.yml]: I wasn't sure, so I put an integer 😅

File: aio/aio-proxy/aio_proxy/doc/open-api.yml (hunk 5 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/doc/open-api.yml]: I do not use the `indicederepetition` column, so I typed it as string just in case.

**Comment by HAEKADI** [comment]
I am going to merge to test the production today. Any additional modifications and changes to urls etc will be felt with in another PR.

**Pull request merged**